src/main/java/org/sonarlint/intellij/ui/SonarLintRulePanel.kt

```diff
@@ -172,15 +178,15 @@ class SonarLintRulePanel(private val project: Project) : JBLoadingPanel(BorderLa
         tabDesc.content.map({ nonContextual -> htmlViewer.updateHtml(nonContextual.htmlContent) }, { contextual ->
             val comboPanel = JBPanel<JBPanel<*>>(HorizontalLayout(JBUI.scale(UIUtil.DEFAULT_HGAP)))
             comboPanel.add(JBLabel("Which component or framework contains the issue?"))
-            val contextCombo = ComboBox(DefaultComboBoxModel(contextual.toTypedArray()))
+            val contextCombo = ComboBox(DefaultComboBoxModel(contextual.contextualSections.toTypedArray()))
             contextCombo.renderer = SimpleListCellRenderer.create("", ActiveRuleContextualSectionDto::getDisplayName)
             contextCombo.addActionListener {
                 val htmlContent = (contextCombo.selectedItem as ActiveRuleContextualSectionDto).htmlContent
                 htmlViewer.updateHtml(htmlContent)
             }
             comboPanel.add(contextCombo)
             sectionPanel.add(comboPanel, BorderLayout.NORTH)
-            contextCombo.selectedIndex = 0
+            contextCombo.selectedIndex = contextual.contextualSections.size - 1
```

Comment:
Here you are assuming the default context is always last in the list. But we added the attribute `contextual.defaultContextKey` precisely to avoid this "magic constant" on the client side.
Should be something like:
```suggestion
            contextCombo.selectedIndex = contextual.contextualSections.indexOf(section -> section..key == contextual.defaultContextKey).orElse(0)
```